Utils/ClanUtils.cs

```diff
@@ -17,5 +50,212 @@ public static float RecalculateClanPP(this AppContext context, int clanId)
+                {
+                    var updateClan = clanRanking.Where(cr => cr.Clan == clan.Key).FirstOrDefault();
+                    if (updateClan != null)
+                    {
+                        updateClan.LastUpdateTime = clan.Value.lastUpdateTime.ToString();
+                        updateClan.ClanPP = clan.Value.clanPP;
+                        updateClan.ClanRank = clanRankCounter;
+                        updateClan.ClanAverageRank = clan.Value.totalRank / clan.Value.numberOfScores;
+                        updateClan.ClanAverageAccuracy = clan.Value.totalAcc / clan.Value.numberOfScores;
+                        updateClan.ClanTotalScore = clan.Value.totalScore;
+                        updateClan.AssociatedScores = clan.Value.Scores;
+                    }
+                    else
+                    {
+                        clanRanking.Add(new ClanRanking { 
+                            Clan = clan.Key,
+                            LastUpdateTime = clan.Value.lastUpdateTime.ToString(),
+                            ClanPP = clan.Value.clanPP,
+                            ClanRank = clanRankCounter,
+                            ClanAverageRank = clan.Value.totalRank / clan.Value.numberOfScores,
+                            ClanAverageAccuracy = clan.Value.totalAcc / clan.Value.numberOfScores,
+                            ClanTotalScore = clan.Value.totalScore,
+                            LeaderboardId = leaderboard.Id,
+                            Leaderboard = leaderboard,
+                            AssociatedScores = clan.Value.Scores
+                        });
+                    }
+                    clanRankCounter++;
+                }
+            }
+
+            return leaderboard.ClanRanking;
+        }
+
+        private static void AddCapturedLeaderboard(
+            ref Dictionary<Clan, ClanRankingData> newClanRankingData,
+            Leaderboard leaderboard,
+            AppContext context)
+        {
+            // Add leaderboard to new captor
+            if (newClanRankingData.First().Key.CapturedLeaderboards == null)
+            {
+                newClanRankingData.First().Key.CapturedLeaderboards = new List<Leaderboard>
+                {
+                    leaderboard
+                };
+            }
+            else
+            {
+                // Check to make sure the clan hasn't already captured this map
+                if (!newClanRankingData.First().Key.CapturedLeaderboards.Contains(leaderboard))
+                {
+                    newClanRankingData.First().Key.CapturedLeaderboards.Add(leaderboard);
+                }
+            }
+
+            // Calculate the number of ranked maps, I feel like this should be a static global or something
+            // We will use this to tell what % of the entire ranked map pool a clan has captured.
+            int rankedMapCount = context
+                .Leaderboards
+                .Include(lb => lb.Difficulty)
```

Comment:
Don't need to use `Include` for `Where` with FKs. Includes are needed only for queries to retrieve the data without explicit `Select`.
For example `Players.Where(p => p.Clans.FirstOrDefault(blabla))` or `Players.Select(p => new { ClanCount = p.Clans.Count() })` will not require `Include clans`, but `player = context.Players.Some; player.Clans.blabla` will.

Reply:
Also retrieving the ranked map count on every call of this function is a bit expensive as your noticed. Could add this to RankingService on start as static variable

Reply:
Does "on start" mean in constructor for the rankingService? Should that variable be updated in RankMaps() or any other location?
Looks like RankingService is only created from Startup.cs, and I'm not sure how often the production server resets.

Reply:
RankingService has an `ExecuteAsync` function which is called on server startup. It then sets a timer for 10am of Friday to rank maps. You can calculate the initial count in `ExecuteAsync` and then update it in `RankMaps` after new maps are ranked. You will need to update clan ranking for them anyway.

Reply:
See https://github.com/BeatLeader/beatleader-server/pull/36#discussion_r1310947615
Implements RankedMapCount in RankingService